synapse/replication/tcp/handler.py

```diff
@@ -352,24 +352,33 @@ def start_replication(self, hs: "HomeServer") -> None:
 
         reactor = hs.get_reactor()
         redis_config = hs.config.redis
-        if hs.config.redis.redis_use_tls:
-            ssl_context_factory = ClientContextFactory(hs.config.redis)
-            reactor.connectSSL(
-                redis_config.redis_host,
-                redis_config.redis_port,
+        if redis_config.redis_path is not None:
+            reactor.connectUNIX(
+                redis_config.redis_path,
                 self._factory,
-                ssl_context_factory,
                 timeout=30,
-                bindAddress=None,
+                checkPID=False,
             )
+
         else:
-            reactor.connectTCP(
-                redis_config.redis_host,
-                redis_config.redis_port,
-                self._factory,
-                timeout=30,
-                bindAddress=None,
-            )
+            if hs.config.redis.redis_use_tls:
```

Comment:
Probably do an `elif` here just to keep the indentation smaller.

Reply:
Fixed in [ce11791](https://github.com/matrix-org/synapse/pull/15644/commits/ce1179146fdeb7a66d0dd9afdad4da11e917e0d1)